### DISCUSSION TITLE
Please help me

### DISCUSSION THREAD

**Discussion by Bee1130** [nextauthjs/next-auth#9853] [Help] [answered]
Hello everyone.
I am making some next.js project and I use app router.
I tried to use next-auth to implement authentication but I couldn't find the exact guide or sample, example for next.js app router + credentials provider.
Please guide me.
How can I implement it on app router?
Thank-you.

**Comment by juan-carlos-correa** [comment, marked as answer]
The docs warns about using the credentials provider because of the inherited complexity, so there is not a full example at all.

But I did it in a personal project successfully, using email and password. But I finished removing it because it was an unnecessary cost with a minimal benefit in my case. Probably it can help you.

Here is the repo: https://github.com/product-makers-hub/open-micro-saas

You would need to do a checkout to this commit: https://github.com/product-makers-hub/open-micro-saas/commit/311977fd8de43f7292182e46f1a53dad8c5bfb69

If I'm correct, it would be this, once you cloned the repo:

```
git checkout 311977f
```
I hope it helps :D

**Comment by confidentmeerkat** [comment]
The docs warns about using the credentials provider because of the inherited complexity, so there is not a full example at all.

But I did it in a personal project successfully, using email and password. But I finished removing it because it was an unnecessary cost with a minimal benefit in my case. Probably it can help you.

Here is the repo: https://github.com/product-makers-hub/open-micro-saas

You would need to do a checkout to this commit: https://github.com/product-makers-hub/open-micro-saas/commit/311977fd8de43f7292182e46f1a53dad8c5bfb69

If I'm correct, it would be this, once you cloned the repo:

git checkout 311977f
I hope it helps :D
